src/PublicApiAnalyzers/Core/CodeFixes/DeclarePublicApiFix.cs

```diff
@@ -295,10 +295,14 @@ protected override async Task<Solution> GetChangedSolutionAsync(CancellationToke
                     newSolution = newSolution.WithAdditionalDocumentText(pair.Key, pair.Value);
                 }
 
+                using var uniqueProjectPaths = PooledHashSet<string>.GetInstance();
                 foreach (KeyValuePair<ProjectId, SourceText> pair in addedPublicSurfaceAreaText)
                 {
                     var project = newSolution.GetProject(pair.Key);
-                    newSolution = AddPublicApiFiles(project, pair.Value);
+                    if (project.FilePath != null && uniqueProjectPaths.Add(project.FilePath))
```

Comment:
Is it possible to somehow create a test reproducing the issue so that it's not regressed later on? If not, could you add a note in the src?

Reply:
Good idea, I added a comment. I don't know of a good way of creating a unit test with multi-tfm targets.